packages/gatsby/src/redux/actions/public.js

```diff
@@ -416,6 +415,17 @@ ${reservedFields.map(f => `  * "${f}"`).join(`\n`)}
     if (page.defer) {
       pageMode = `DSR`
     }
+
+    // TODO move to AST Check
+    const fileContent = fs.readFileSync(page.component).toString()
```

Comment:
feels expensive without any memoization / cache

Reply:
👍 this isjust to make it work